src/plugins/data/common/es_query/kuery/ast/kuery.peg

```diff
@@ -209,7 +209,7 @@ Literal "literal"
   = QuotedString / UnquotedLiteral
 
 QuotedString
-  = '"' prefix:QuotedCharacter* cursor:Cursor suffix:QuotedCharacter* '"' {
+  = &{ return parseCursor; } '"' prefix:QuotedCharacter* cursor:Cursor suffix:QuotedCharacter* '"' {
```

Comment:
See https://pegjs.org/documentation#grammar-syntax-and-semantics. This is a "predicate" which essentially does the same check as above before trying this grammar rule.